cloudhub/modules/ROOT/pages/app-migration.adoc

```diff
@@ -0,0 +1,142 @@
++
+image::traffic-switching.png[CloudHub to CloudHub 2.0 app traffic switching window.]
+
+[NOTE]
+You can roll back to CloudHub if there are failures during this stage.
+
+=== Understanding Traffic Source
+
+The traffic source determines which type of DNS record are updated to manage the traffic flow between your CloudHub and CloudHub 2.0 applications during the migration process. Understanding this concept helps you anticipate how ‌‌traffic is managed based on your application's existing and new endpoint configurations.
+
+CloudHub can theoretically receive traffic from two main sources:
+
+* Shared Load Balancer (SLB): This is typically associated with the default endpoints of your CloudHub application, often listening on ports `8081/8082`.
+* Dedicated Load Balancer (DLB): This is usually associated with vanity domains configured for your CloudHub application, often listening on ports `8091/8092`.
+
+CloudHub 2.0 applications, in contrast, only listen on a single port, `8081`. 
+
+During traffic switching, CloudHub needs to know the traffic source to correctly modify the DNS records. This is because applications using SLB have a different DNS record structure than those using DLB.
+
+To facilitate the gradual migration of traffic from CloudHub to CloudHub 2.0, the system creates weighted DNS records. These records direct a certain percentage of traffic to your CloudHub application and the remaining percentage to your linked CloudHub 2.0 application. The specific DNS record that needs to be weighted depends on whether the traffic to your CloudHub application originates from the SLB or the DLB.
+
+The system infers the traffic source based on the endpoints configured for your application. The rules for inference are:
+
+* If your application has a CloudHub default endpoint, the SLB DNS record is used for traffic switching. CloudHub default domains follow patterns like `$region.$env.cloudhub.io` or `*.ca-c1.stgx.cloudhub.io/`.
+* If your application has any other endpoints, such as vanity domains or mule-worker endpoints, the DLB DNS record is used for traffic switching. 
+* Vanity domains are any domains other than the default CloudHub or CloudHub 2.0 domains. Make sure that the vanity domain on CloudHub 2.0 matches the one on the CloudHub application, or avoid setting a traffic switch percentage to prevent dropped traffic.
+* If your application has both default domains and vanity domains configured, default domains take precedence over vanity domains.
+
+If you change your application's endpoint configuration, the inferred traffic source can also change. For example, if you initially had only vanity domains (DLB) and then add a CloudHub default endpoint, the traffic source is inferred as SLB.
+
+=== Managing API Policies During Traffic Switching
+
+When API policies are applied to your application in the new runtime, they begin with a clean internal status.
+
+==== Policies Unchanged by the Runtime:
```

Comment:
What does Runtime mean in this context?

Reply:
My guess is it refers to the app's active/running time in the new target (CH2). I asked an SME to confirm.  Once I get the answer, I'll change it to make it clearer.